src/Gateway.php

```diff
@@ -116,63 +114,118 @@ public function payment_fields() {
+		$additional_fields = array(
+			'company_number'              => array(
+				'label' => __( 'Company number', 'ledyer-payments-for-woocommerce' ),
+				'type'  => 'text',
+				'show'  => true,
+			),
+			'ledyer_customer_reference_1' => array(
+				'id'    => '_ledyer_customer_reference_1',
+				'label' => __( 'Customer reference 1', 'ledyer-payments-for-woocommerce' ),
+				'type'  => 'text',
+				'show'  => true,
+				'value' => $order->get_meta( '_ledyer_customer_reference_1' ),
+			),
+			'ledyer_customer_reference_2' => array(
+				'id'    => '_ledyer_customer_reference_2',
+				'label' => __( 'Customer reference 2', 'ledyer-payments-for-woocommerce' ),
+				'type'  => 'text',
+				'show'  => true,
+				'value' => $order->get_meta( '_ledyer_customer_reference_2' ),
+			),
 		);
 
-		$fields['billing_customer_reference2'] = array(
-			'label'       => __( 'Reference 2', 'ledyer-payments-for-woocommerce' ),
-			'placeholder' => __( 'For example, purchase order number', 'ledyer-payments-for-woocommerce' ),
-			'required'    => false,
-			'class'       => array( 'form-row-wide' ),
-			'clear'       => true,
-			'priority'    => 35,
+		// Inserts the additional fields after the 'company' field.
+		$index = array_search(
+			'company',
+			array_keys( $billing_fields ),
+			true
 		);
 
-		return $fields;
+		// Increment $index by 1 to include the first element too (since $index starts from 0, but length starts from 1).
+		$before = array_slice( $billing_fields, 0, $index + 1 );
+		$after  = array_slice( $billing_fields, $index );
+		return $before + $additional_fields + $after;
 	}
 
 	/**
-	 * Add additional billing fields to the admin order page.
+	 * Process the admin order update.
 	 *
-	 * @param array $fields The billing fields.
-	 * @return array
+	 * @param int $order_id The WooCommerce order id.
+	 * @return void
 	 */
-	public function additional_admin_billing_fields( $fields ) {
-		$fields['customer_reference1'] = array(
-			'label' => __( 'Customer reference 1', 'ledyer-payments-for-woocommerce' ),
-			'show'  => true,
-		);
+	public function admin_order_processing( $order_id ) {
+		$order = wc_get_order( $order_id );
+		if ( ! $this->is_chosen_gateway( $order ) ) {
+			return;
+		}
 
-		$fields['customer_reference2'] = array(
-			'label' => __( 'Customer reference 2', 'ledyer-payments-for-woocommerce' ),
-			'show'  => true,
-		);
+		// Notice that the field name is prefixed with 'billing_' unless an 'id' is explicitly set which it is in our case. See additional_admin_billing_fields() in this file.
+		$customer_reference_1 = filter_input( INPUT_POST, '_ledyer_customer_reference_1', FILTER_SANITIZE_FULL_SPECIAL_CHARS );
+		if ( isset( $customer_reference_1 ) ) {
+			$stored_reference_1 = $order->get_meta( '_ledyer_customer_reference_1' );
+			if ( $customer_reference_1 !== $stored_reference_1 ) {
```

Comment:
Im not sure we need to add these extra checks. WooCommerce will update the order anyway if im not misstaken, so it feels like its not really needed. So we can remove the extra check and getting the metadata for both this and reference 2